cvat/apps/engine/views.py

```diff
@@ -1275,24 +1277,75 @@ def content(self, request, pk):
+                raise FileNotFoundError(errno.ENOENT,
+                    "Not found on the cloud storage {}".format(db_storage.display_name), manifest_path)
+
+            full_manifest_path = os.path.join(db_storage.get_storage_dirname(), manifest_path)
+            if not os.path.exist(full_manifest_path) or \
+                    os.path.getmtime(full_manifest_path) < storage.get_file_last_modified(full_manifest_path):
+                # TODO: create sub dirs
+                storage.download_file(manifest_path, full_manifest_path)
+                manifest = ImageManifestManager(full_manifest_path)
+                # need to reset previon index
+                manifest.reset_index()
                 manifest.init_index()
                 manifest_files = manifest.data
-            content = {f:[] for f in set(storage_files) | set(manifest_files)}
-            for key, _ in content.items():
-                if key in storage_files: content[key].append('s') # storage
-                if key in manifest_files: content[key].append('m') # manifest
+            return Response(data=manifest_files, content_type="text/plain")
 
-            data = json.dumps(content)
-            return Response(data=data, content_type="aplication/json")
+        except CloudStorageModel.DoesNotExist:
+            message = f"Storage {pk} does not exist"
+            slogger.glob.error(message)
+            return HttpResponseNotFound(message)
+        except FileNotFoundError as ex:
+            msg = f"{ex.strerror} {ex.filename}"
+            slogger.cloud_storage[pk].info(msg)
+            return Response(data=msg, status=status.HTTP_404_NOT_FOUND)
+        except Exception as ex:
+            return HttpResponseBadRequest(str(ex))
 
+    @swagger_auto_schema(
+        method='get',
+        operation_summary='Method returns a preview image from a cloud storage',
+        responses={
+            '200': openapi.Response(description='Preview'),
+        },
+        tags=['cloud storages']
+    )
+    @action(detail=True, methods=['GET'], url_path='preview')
+    def preview(self, request, pk):
+        try:
+            db_storage = CloudStorageModel.objects.get(pk=pk)
+            if not os.path.exists(db_storage.get_preview_path()):
+                credentials = Credentials()
+                credentials.convert_from_db({
+                    'type': db_storage.credentials_type,
+                    'value': db_storage.credentials,
+                })
+                details = {
+                    'resource': db_storage.resource,
+                    'credentials': credentials,
+                    'specific_attributes': db_storage.get_specific_attributes()
+                }
+                storage = get_cloud_storage_instance(cloud_provider=db_storage.provider_type, **details)
+                storage.initialize_content()
+                storage_images = [f for f in storage.content if MEDIA_TYPES['image']['has_mime_type'](f)]
```

Comment:
Should not we use images only from manifest file?

Reply:
No, In fact, the manifest file in cloud storage and the manifest required for the task are different files.

Reply:
I mean images from cloud storages manifest file

Reply:
Although, there can be several manifests on the cloud storage, then the choice of one or another occurs at the stage of creating a task. When creating and when working with cloud storage, the data manifest is not used and, accordingly, is not downloaded.